### ISSUE TITLE
Rails 6 app does not render templates when started from a Dropbox folder

### ISSUE THREAD

**Issue by catswithbenefits** [rails/rails#37719]
### Steps to reproduce
```

myuser@192 Dropbox % pwd
/Users/myuser/Dropbox
myuser@192 Dropbox % rails new bugreport
myuser@192 Dropbox % cd bugreport 
myuser@192 bugreport % bundle
myuser@192 bugreport % rails g controller testc index
myuser@192 bugreport % rails s

http://localhost:3000/ # Yay! You’re on Rails!

http://localhost:3000/testc/index 
# No template for interactive request
# TestcController#index is missing a template for request formats: text/html

Exit Server

myuser@192 bugreport % cd ..
myuser@192 Dropbox % mv bugreport ../
myuser@192 Dropbox % cd ../bugreport 
myuser@192 bugreport % pwd
/Users/myuser/bugreport
myuser@192 bugreport % rails s

http://localhost:3000/testc/index 
# Testc#index
# Find me in app/views/testc/index.html.erb

```

### Expected behavior
The app should work independant from the folder it is started from. Just like it did in 5.2 which is not affected by this issue.

### Actual behavior
Rendering any ERB view fails.

Log output:
```

`Completed 406 Not Acceptable in 5ms (ActiveRecord: 0.0ms | Allocations: 1970)
 
ActionController::MissingExactTemplate (TestcController#index is missing a template for request formats: text/html):`
```

### System configuration
**Rails version**:
6

**Ruby version**:
ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]

**Operating System**
Mac Os Catalina

**Comment by kassio** [comment]
I tried with Rails 6.0.1 and couldn't replicate, can you try again with Rails 6.0.1, please?

**Comment by catswithbenefits** [comment]
I can confirm the problem persists with 6.0.1

Removing git from the folder does not change the faulty behaviour. 
Besides, Rails 5.2 is not affected by this issue, so this is a regression caused by Rails 6.


```
bugreport % cat Gemfile|grep "gem 'rails'"
gem 'rails', '~> 6.0.1'

bugreport % rails -v
Rails 6.0.1

% git fsck --full
Checking object directories: 100% (256/256), done.
notice: HEAD points to an unborn branch (master)
notice: No default references

bugreport % rm -rf .git 
bugreport % git status
fatal: not a git repository (or any of the parent directories): .git

# http://localhost:3000/testc/index
# No template for interactive request
# TestcController#index is missing a template for request formats: text/html
```

**Comment by huseyin** [comment]
@catswithbenefits I tried to reproduce the issue with Rails 6 and couldn't replicate. Everything looks good with Dropbox. Something in your local may be causing this error. Can you isolate the issue from your local development environment and try again?

**Comment by catswithbenefits** [comment]
Please let me know what you want me to check.

The error happens with an existing app as well as with a fresh “rails new bugreport” app. There aren’t any issues with my system that I’m aware of.

**Comment by huseyin** [comment]
@catswithbenefits I can't reproduce the issue. I followed the steps you specified in the issue body, Rails doesn't shout out with both 6 and 6.0.1 versions. This seems to be an issue based on Dropbox or development environment.

**Comment by huseyin** [comment]
@catswithbenefits any progress on this?

**Comment by catswithbenefits** [comment]
Yes:

Meanwhile I got a new Macbook. I transfered all my files and settings from the old system with Apple's migration assistant. I also uninstalled & deleted Dropbox and installed it again.

The problem persists with the new computer. 

I have no idea how to debug that issue and I'm now developing my Rails app outside the Dropbox folder.

**Comment by RogerPodacter** [comment]
FWIW I'm having the exact same problem.

**Comment by paluchas** [comment]
Hello, I hit nearly the same error. This is not related to the Dropbox folder as originally reported:

__ActionController::MissingExactTemplate (XXX#index is missing a template for request formats: text/html):__

My dev is MacOS Catalina 10.15.2 with __ruby-install__ and __chruby__ as installed by __brew__.  I have followed this [Rails API CRUD + React tutorial ](https://hibbard.eu/rails-react-crud-app/). I initiated my project environment with Rails 6.0.2.1 and Ruby 2.7.0 when the error shows as:
```
Puma starting in single mode...
* Version 4.3.1 (ruby 2.7.0-p0), codename: Mysterious Traveller
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://0.0.0.0:3000
Use Ctrl-C to stop
...
Processing by SiteController#index as HTML
...
ActionController::MissingExactTemplate (SiteController#index is missing a template for request formats: text/html):

actionpack (6.0.2.1) lib/action_controller/metal/implicit_render.rb:45:in `default_render'
actionpack (6.0.2.1) lib/action_controller/metal/basic_implicit_render.rb:6:in `block in send_action'
actionpack (6.0.2.1) lib/action_controller/metal/basic_implicit_render.rb:6:in `tap'
actionpack (6.0.2.1) lib/action_controller/metal/basic_implicit_render.rb:6:in `send_action'
actionpack (6.0.2.1) lib/abstract_controller/base.rb:196:in `process_action'
actionpack (6.0.2.1) lib/action_controller/metal/rendering.rb:30:in `process_action'
actionpack (6.0.2.1) lib/abstract_controller/callbacks.rb:42:in `block in process_action'
activesupport (6.0.2.1) lib/active_support/callbacks.rb:135:in `run_callbacks'
actionpack (6.0.2.1) lib/abstract_controller/callbacks.rb:41:in `process_action'
actionpack (6.0.2.1) lib/action_controller/metal/rescue.rb:22:in `process_action'
actionpack (6.0.2.1) lib/action_controller/metal/instrumentation.rb:33:in `block in process_action'
activesupport (6.0.2.1) lib/active_support/notifications.rb:180:in `block in instrument'
activesupport (6.0.2.1) lib/active_support/notifications/instrumenter.rb:24:in `instrument'
activesupport (6.0.2.1) lib/active_support/notifications.rb:180:in `instrument'
actionpack (6.0.2.1) lib/action_controller/metal/instrumentation.rb:32:in `process_action'
actionpack (6.0.2.1) lib/action_controller/metal/params_wrapper.rb:245:in `process_action'
activerecord (6.0.2.1) lib/active_record/railties/controller_runtime.rb:27:in `process_action'
actionpack (6.0.2.1) lib/abstract_controller/base.rb:136:in `process'
actionview (6.0.2.1) lib/action_view/rendering.rb:39:in `process'
actionpack (6.0.2.1) lib/action_controller/metal.rb:191:in `dispatch'
actionpack (6.0.2.1) lib/action_controller/metal.rb:252:in `dispatch'
actionpack (6.0.2.1) lib/action_dispatch/routing/route_set.rb:51:in `dispatch'
actionpack (6.0.2.1) lib/action_dispatch/routing/route_set.rb:33:in `serve'
actionpack (6.0.2.1) lib/action_dispatch/journey/router.rb:49:in `block in serve'
actionpack (6.0.2.1) lib/action_dispatch/journey/router.rb:32:in `each'
actionpack (6.0.2.1) lib/action_dispatch/journey/router.rb:32:in `serve'
actionpack (6.0.2.1) lib/action_dispatch/routing/route_set.rb:837:in `call'
rack (2.0.8) lib/rack/tempfile_reaper.rb:15:in `call'
rack (2.0.8) lib/rack/etag.rb:25:in `call'
rack (2.0.8) lib/rack/conditional_get.rb:25:in `call'
rack (2.0.8) lib/rack/head.rb:12:in `call'
actionpack (6.0.2.1) lib/action_dispatch/http/content_security_policy.rb:18:in `call'
rack (2.0.8) lib/rack/session/abstract/id.rb:259:in `context'
rack (2.0.8) lib/rack/session/abstract/id.rb:253:in `call'
actionpack (6.0.2.1) lib/action_dispatch/middleware/cookies.rb:648:in `call'
activerecord (6.0.2.1) lib/active_record/migration.rb:567:in `call'
actionpack (6.0.2.1) lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
activesupport (6.0.2.1) lib/active_support/callbacks.rb:101:in `run_callbacks'
actionpack (6.0.2.1) lib/action_dispatch/middleware/callbacks.rb:26:in `call'
actionpack (6.0.2.1) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (6.0.2.1) lib/action_dispatch/middleware/actionable_exceptions.rb:17:in `call'
actionpack (6.0.2.1) lib/action_dispatch/middleware/debug_exceptions.rb:32:in `call'
web-console (4.0.1) lib/web_console/middleware.rb:132:in `call_app'
web-console (4.0.1) lib/web_console/middleware.rb:28:in `block in call'
web-console (4.0.1) lib/web_console/middleware.rb:17:in `catch'
web-console (4.0.1) lib/web_console/middleware.rb:17:in `call'
actionpack (6.0.2.1) lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
railties (6.0.2.1) lib/rails/rack/logger.rb:38:in `call_app'
railties (6.0.2.1) lib/rails/rack/logger.rb:26:in `block in call'
activesupport (6.0.2.1) lib/active_support/tagged_logging.rb:80:in `block in tagged'
activesupport (6.0.2.1) lib/active_support/tagged_logging.rb:28:in `tagged'
activesupport (6.0.2.1) lib/active_support/tagged_logging.rb:80:in `tagged'
railties (6.0.2.1) lib/rails/rack/logger.rb:26:in `call'
sprockets-rails (3.2.1) lib/sprockets/rails/quiet_assets.rb:13:in `call'
actionpack (6.0.2.1) lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
actionpack (6.0.2.1) lib/action_dispatch/middleware/request_id.rb:27:in `call'
rack (2.0.8) lib/rack/method_override.rb:22:in `call'
rack (2.0.8) lib/rack/runtime.rb:22:in `call'
activesupport (6.0.2.1) lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
actionpack (6.0.2.1) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (6.0.2.1) lib/action_dispatch/middleware/static.rb:126:in `call'
rack (2.0.8) lib/rack/sendfile.rb:111:in `call'
actionpack (6.0.2.1) lib/action_dispatch/middleware/host_authorization.rb:83:in `call'
webpacker (4.2.2) lib/webpacker/dev_server_proxy.rb:23:in `perform_request'
rack-proxy (0.6.5) lib/rack/proxy.rb:57:in `call'
railties (6.0.2.1) lib/rails/engine.rb:526:in `call'
puma (4.3.1) lib/puma/configuration.rb:228:in `call'
puma (4.3.1) lib/puma/server.rb:681:in `handle_request'
puma (4.3.1) lib/puma/server.rb:472:in `process_client'
puma (4.3.1) lib/puma/server.rb:328:in `block in run'
puma (4.3.1) lib/puma/thread_pool.rb:134:in `block in spawn_thread'
```
The error persist with __ruby 2.6.5__. However, once Rails gem dongraded to __Rails 5.2__ it works ok.

Here is my __bundle env__ as original error shows

## Environment

```
Bundler       2.1.2
  Platforms   ruby, x86_64-darwin-19
Ruby          2.7.0p0 (2019-12-25 revision 647ee6f091eafcce70ffb75ddf7e121e192ab217) [x86_64-darwin19]
  Full Path   /Users/paluchas/.rubies/ruby-2.7.0/bin/ruby
  Config Dir  /Users/paluchas/.rubies/ruby-2.7.0/etc
RubyGems      3.1.2
  Gem Home    /Users/paluchas/.gem/ruby/2.7.0
  Gem Path    /Users/paluchas/.gem/ruby/2.7.0:/Users/paluchas/.rubies/ruby-2.7.0/lib/ruby/gems/2.7.0
  User Home   /Users/paluchas
  User Path   /Users/paluchas/.gem/ruby/2.7.0
  Bin Dir     /Users/paluchas/.gem/ruby/2.7.0/bin
Tools         
  Git         2.21.0 (Apple Git-122.2)
  RVM         not installed
  rbenv       not installed
  chruby      0.3.9
```

## Bundler Build Metadata

```
Built At          2019-12-30
Git SHA           unknown
Released Version  false
```

## Bundler settings

```
path
  Set for the current user (/Users/paluchas/.bundle/config): "vendor/bundle"
without
  Set for your local app (/Users/paluchas/current/rails_api+react/.bundle/config): []
  Set for the current user (/Users/paluchas/.bundle/config): [:development, :test]
# bundle_build.libv8
  You have not configured a value for `# bundle_build.libv8`
# bundle_build.therubyracer
  You have not configured a value for `# bundle_build.therubyracer`
```

## Gemfile

### Gemfile

```ruby
source 'https://rubygems.org'
git_source(:github) { |repo| "https://github.com/#{repo}.git" }

ruby '2.7.0'

# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
# gem 'rails', '~> 5.2.1'
gem 'rails', '~> 6.0.2', '>= 6.0.2.1'
# Use sqlite3 as the database for Active Record
gem 'sqlite3', '~> 1.4'
# Use Puma as the app server
gem 'puma', '~> 4.1'
# Use responder to DRY API controllers
gem 'responders'
# Use SCSS for stylesheets
gem 'sass-rails', '>= 6'
# Transpile app-like JavaScript. Read more: https://github.com/rails/webpacker
gem 'webpacker', '~> 4.0'
# Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
gem 'turbolinks', '~> 5'
# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
gem 'jbuilder', '~> 2.7'
# Use Redis adapter to run Action Cable in production
# gem 'redis', '~> 4.0'
# Use Active Model has_secure_password
# gem 'bcrypt', '~> 3.1.7'

# Use Active Storage variant
# gem 'image_processing', '~> 1.2'

# Reduces boot times through caching; required in config/boot.rb
gem 'bootsnap', '>= 1.4.2', require: false

group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
end

group :development do
  # Access an interactive console on exception pages or by calling 'console' anywhere in the code.
  gem 'web-console', '>= 3.3.0'
  gem 'listen', '>= 3.0.5', '< 3.2'
  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
  gem 'spring'
  gem 'spring-watcher-listen', '~> 2.0.0'
  gem 'solargraph'
  gem 'rubocop'
end

group :test do
  # Adds support for Capybara system testing and selenium driver
  gem 'capybara', '>= 2.15'
  gem 'selenium-webdriver'
  # Easy installation and use of web drivers to run system tests with browsers
  gem 'webdrivers'
end

# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
```

### Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    actioncable (6.0.2.1)
      actionpack (= 6.0.2.1)
      nio4r (~> 2.0)
      websocket-driver (>= 0.6.1)
    actionmailbox (6.0.2.1)
      actionpack (= 6.0.2.1)
      activejob (= 6.0.2.1)
      activerecord (= 6.0.2.1)
      activestorage (= 6.0.2.1)
      activesupport (= 6.0.2.1)
      mail (>= 2.7.1)
    actionmailer (6.0.2.1)
      actionpack (= 6.0.2.1)
      actionview (= 6.0.2.1)
      activejob (= 6.0.2.1)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (6.0.2.1)
      actionview (= 6.0.2.1)
      activesupport (= 6.0.2.1)
      rack (~> 2.0, >= 2.0.8)
      rack-test (>= 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.2.0)
    actiontext (6.0.2.1)
      actionpack (= 6.0.2.1)
      activerecord (= 6.0.2.1)
      activestorage (= 6.0.2.1)
      activesupport (= 6.0.2.1)
      nokogiri (>= 1.8.5)
    actionview (6.0.2.1)
      activesupport (= 6.0.2.1)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.1, >= 1.2.0)
    activejob (6.0.2.1)
      activesupport (= 6.0.2.1)
      globalid (>= 0.3.6)
    activemodel (6.0.2.1)
      activesupport (= 6.0.2.1)
    activerecord (6.0.2.1)
      activemodel (= 6.0.2.1)
      activesupport (= 6.0.2.1)
    activestorage (6.0.2.1)
      actionpack (= 6.0.2.1)
      activejob (= 6.0.2.1)
      activerecord (= 6.0.2.1)
      marcel (~> 0.3.1)
    activesupport (6.0.2.1)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 0.7, < 2)
      minitest (~> 5.1)
      tzinfo (~> 1.1)
      zeitwerk (~> 2.2)
    addressable (2.7.0)
      public_suffix (>= 2.0.2, < 5.0)
    ast (2.4.0)
    backport (1.1.2)
    bindex (0.8.1)
    bootsnap (1.4.5)
      msgpack (~> 1.0)
    builder (3.2.4)
    byebug (11.0.1)
    capybara (3.30.0)
      addressable
      mini_mime (>= 0.1.3)
      nokogiri (~> 1.8)
      rack (>= 1.6.0)
      rack-test (>= 0.6.3)
      regexp_parser (~> 1.5)
      xpath (~> 3.2)
    childprocess (3.0.0)
    concurrent-ruby (1.1.5)
    crass (1.0.5)
    erubi (1.9.0)
    ffi (1.11.3)
    globalid (0.4.2)
      activesupport (>= 4.2.0)
    i18n (1.7.0)
      concurrent-ruby (~> 1.0)
    jaro_winkler (1.5.4)
    jbuilder (2.9.1)
      activesupport (>= 4.2.0)
    listen (3.1.5)
      rb-fsevent (~> 0.9, >= 0.9.4)
      rb-inotify (~> 0.9, >= 0.9.7)
      ruby_dep (~> 1.2)
    loofah (2.4.0)
      crass (~> 1.0.2)
      nokogiri (>= 1.5.9)
    mail (2.7.1)
      mini_mime (>= 0.1.1)
    marcel (0.3.3)
      mimemagic (~> 0.3.2)
    maruku (0.7.3)
    method_source (0.9.2)
    mimemagic (0.3.3)
    mini_mime (1.0.2)
    mini_portile2 (2.4.0)
    minitest (5.13.0)
    msgpack (1.3.1)
    nio4r (2.5.2)
    nokogiri (1.10.7)
      mini_portile2 (~> 2.4.0)
    parallel (1.19.1)
    parser (2.7.0.1)
      ast (~> 2.4.0)
    public_suffix (4.0.2)
    puma (4.3.1)
      nio4r (~> 2.0)
    rack (2.0.8)
    rack-proxy (0.6.5)
      rack
    rack-test (1.1.0)
      rack (>= 1.0, < 3)
    rails (6.0.2.1)
      actioncable (= 6.0.2.1)
      actionmailbox (= 6.0.2.1)
      actionmailer (= 6.0.2.1)
      actionpack (= 6.0.2.1)
      actiontext (= 6.0.2.1)
      actionview (= 6.0.2.1)
      activejob (= 6.0.2.1)
      activemodel (= 6.0.2.1)
      activerecord (= 6.0.2.1)
      activestorage (= 6.0.2.1)
      activesupport (= 6.0.2.1)
      bundler (>= 1.3.0)
      railties (= 6.0.2.1)
      sprockets-rails (>= 2.0.0)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.3.0)
      loofah (~> 2.3)
    railties (6.0.2.1)
      actionpack (= 6.0.2.1)
      activesupport (= 6.0.2.1)
      method_source
      rake (>= 0.8.7)
      thor (>= 0.20.3, < 2.0)
    rainbow (3.0.0)
    rake (13.0.1)
    rb-fsevent (0.10.3)
    rb-inotify (0.10.1)
      ffi (~> 1.0)
    regexp_parser (1.6.0)
    responders (3.0.0)
      actionpack (>= 5.0)
      railties (>= 5.0)
    reverse_markdown (1.3.0)
      nokogiri
    rubocop (0.78.0)
      jaro_winkler (~> 1.5.1)
      parallel (~> 1.10)
      parser (>= 2.6)
      rainbow (>= 2.2.2, < 4.0)
      ruby-progressbar (~> 1.7)
      unicode-display_width (>= 1.4.0, < 1.7)
    ruby-progressbar (1.10.1)
    ruby_dep (1.5.0)
    rubyzip (2.0.0)
    sass-rails (6.0.0)
      sassc-rails (~> 2.1, >= 2.1.1)
    sassc (2.2.1)
      ffi (~> 1.9)
    sassc-rails (2.1.2)
      railties (>= 4.0.0)
      sassc (>= 2.0)
      sprockets (> 3.0)
      sprockets-rails
      tilt
    selenium-webdriver (3.142.7)
      childprocess (>= 0.5, < 4.0)
      rubyzip (>= 1.2.2)
    solargraph (0.38.0)
      backport (~> 1.1)
      bundler (>= 1.17.2)
      jaro_winkler (~> 1.5)
      maruku (~> 0.7, >= 0.7.3)
      nokogiri (~> 1.9, >= 1.9.1)
      parser (~> 2.3)
      reverse_markdown (~> 1.0, >= 1.0.5)
      rubocop (~> 0.52)
      thor (~> 0.19, >= 0.19.4)
      tilt (~> 2.0)
      yard (~> 0.9)
    spring (2.1.0)
    spring-watcher-listen (2.0.1)
      listen (>= 2.7, < 4.0)
      spring (>= 1.2, < 3.0)
    sprockets (4.0.0)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.2.1)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    sqlite3 (1.4.2)
    thor (0.20.3)
    thread_safe (0.3.6)
    tilt (2.0.10)
    turbolinks (5.2.1)
      turbolinks-source (~> 5.2)
    turbolinks-source (5.2.0)
    tzinfo (1.2.6)
      thread_safe (~> 0.1)
    unicode-display_width (1.6.0)
    web-console (4.0.1)
      actionview (>= 6.0.0)
      activemodel (>= 6.0.0)
      bindex (>= 0.4.0)
      railties (>= 6.0.0)
    webdrivers (4.2.0)
      nokogiri (~> 1.6)
      rubyzip (>= 1.3.0)
      selenium-webdriver (>= 3.0, < 4.0)
    webpacker (4.2.2)
      activesupport (>= 4.2)
      rack-proxy (>= 0.6.1)
      railties (>= 4.2)
    websocket-driver (0.7.1)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.4)
    xpath (3.2.0)
      nokogiri (~> 1.8)
    yard (0.9.20)
    zeitwerk (2.2.2)

PLATFORMS
  ruby

DEPENDENCIES
  bootsnap (>= 1.4.2)
  byebug
  capybara (>= 2.15)
  jbuilder (~> 2.7)
  listen (>= 3.0.5, < 3.2)
  puma (~> 4.1)
  rails (~> 6.0.2, >= 6.0.2.1)
  responders
  rubocop
  sass-rails (>= 6)
  selenium-webdriver
  solargraph
  spring
  spring-watcher-listen (~> 2.0.0)
  sqlite3 (~> 1.4)
  turbolinks (~> 5)
  tzinfo-data
  web-console (>= 3.3.0)
  webdrivers
  webpacker (~> 4.0)

RUBY VERSION
   ruby 2.7.0p0

BUNDLED WITH
   2.1.2
```

**Comment by mhaksu** [comment]
+1  same exact issue

**Comment by alexander-alyoshin** [comment]
+1

**Comment by rempargo** [comment]
Dropbox Business users tend to have a space in the Dropbox folder name.

    ~/Dropbox (CompanyName)

This can result in problems with some ruby gems and also with Rails that uses an absolute file paths, but are not working correctly with spaces in the path.

You can easily test if this is the case by copying your Rails workfolder and place it onto your Desktop and see if the rails is working correctly. Or place it on another computer with your personal Dropbox (which does not have a space).

'Normal' or 'Individual' Dropbox users, usually just have `~/Dropbox`as folder name, and they will not have this issue. Therefore some Dropbox users are affected and some not.

In the end, it is not a 'Dropbox' issue, but just a filename issue, but the Folder name of Dropbox Business users can't be renamed (raised this with Dropbox support once). 

You can try to make a symlink `ln -s "/Users/yourusername/Dropbox (YourCompany)" "~/Dropbox" 
That used to be my trick before when I encountered similar issues, but it seems that does not work with Rails 6.

I'm on:
Macbook Pro
macOS Mojave
Ruby-2.6.5
Rails 6.0.2.1

**Comment by catswithbenefits** [comment]
@ rempargo : No :-)

Folders with spaces are not the problem. The Dropbox folder is the problem. Even if the path does not contain spaces. 

Besides: There are no error messages and it used to work with Rails 5.

So it's very clear to me that this is a Rails 6 + Dropbox issue and nothing else. 

It's interesting though that you're using the old Mojave version. My gut tells me that one of the new and security features of Catalina is messing up something for rails.

**Comment by rempargo** [comment]
@catswithbenefits Do you use the new Dropbox feature 'SmartSync'? If so, do you have the complete app on 'Local Only' instead of 'Online Only'? (right-mouse click on the folder in the Dropbox context menu).

**Comment by catswithbenefits** [comment]
I'm not using this Dropbox feature.

I first encountered this problem after I upgraded a rails 5 application inside my personal Dropbox folder (without spaces) to rails 6. However also a fresh rails 6 project inside Dropbox has the issue.

Can you test on Catalina? I think the OS and its new security features and more restrictive policies might have something to do with it.

**Comment by rempargo** [comment]
@catswithbenefits You're right, it's still not working (I started rails server from copy on my Desktop), sorry for the confusement. So it is also an issue on my macOS Mojave. 

I checked the working version on my Desktop and the none-working one in Dropbox.
All related files have the same file permissions and file ownership.

I tried Puma (default) as well as Webrick `rails server webrick`, but still ActionController::MissingExactTemplate.

I made a Cloud Sync Dropbox connection on my NAS, and run 'rails server` from this share, but this gave me 'ActionView::Template::Error (Webpacker can't find application in /Volumes/DropboxOnNas/testapp/public/packs/manifest.json. ' not the exact same error message, but also with a template Error.


Macbook Pro
macOS Mojave
Ruby-2.6.5
rvm 1.29.9
Rails 6.0.2.1
Dropbox Business
APFS Volume (see Disk Utility) curious if other with the same problem have APFS
